File: ProEventos.Domain/Evento.cs
namespace ProEventos.Domain;

public class Evento
{
    public int Id { get; set; }
    public string Local { get; set; } = string.Empty;
    public DateTime? DataEvento { get; set; }
    public string Tema { get; set; } = string.Empty;
    public int QtdPessoas { get; set; }
    public string ImagemURL { get; set; } = string.Empty;
    public string Telefone { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public IEnumerable<Lote> Lotes { get; set; } = Enumerable.Empty<Lote>();
    public IEnumerable<RedeSocial> RedesSociais { get; set; } = Enumerable.Empty<RedeSocial>();
    public IEnumerable<PalestranteEvento> PalestrantesEventos { get; set; } = Enumerable.Empty<PalestranteEvento>();
}

File: ProEventos-App/src/app/models/Evento.ts
import { ILote } from "./Lote";
import { RedeSocial } from "./RedeSocial";
import { Palestrante } from "./Palestrante";

export interface Evento {
  id: number;
  local: string;
  dataEvento?: Date;
  tema: string;
  qtdPessoas: number;
  imagemURL: string;
  telefone: string;
  email: string;
  lotes: ILote[];
  redesSociais: RedeSocial[];
  palestrantesEventos: Palestrante[];
}
